test/test_grid.py

```diff
@@ -932,3 +933,9 @@ def test_populate_bounds_GCA_mix(self):
         bounds_xarray = grid.bounds
         face_bounds = bounds_xarray.values
         nt.assert_allclose(grid.bounds.values, expected_bounds, atol=ERROR_TOLERANCE)
+
+    def test_populate_bounds_MPAS(self):
+        xrds = xr.open_dataset(self.gridfile_mpas)
+        uxgrid = ux.Grid.from_dataset(xrds, use_dual=True)
+        bounds_xarray = uxgrid.bounds
+        pass
```

Comment:
This test fails when using `use_dual=False`, which is the hexagonal mesh. This is the behavior I observed with my comment yesterday

When `use_dual=True`, it's a strictly triangular mesh with no fill values. 

> >       return np.array(faces_edges_coordinates)
> E       ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (1791,) + inhomogeneous part.
> 
> uxarray/grid/utils.py:290: ValueError

Reply:
If we want to preserve the fill_values, just simply remove the mask for the face_nodes at each single face. Currently the single_face function will clean up any fill_value. And that's what I wanted it to be. 

But now the entire API structure has changes, then we have to think more about how we want these functions to return. This is the reason why I want this API to be properly handled and don't want to go back to change it again.

Reply:
> Currently the single_face function will clean up any fill_value. And that's what I wanted it to be.

Can you elaborate on this?

Reply:
I was talking about previously, I want the `faces_edges_coordinates` to contain no fill values for each face. Because this `get_cartesian_face_edge_nodes` function was designed to only process 1 face at a time. And that's how I use it in the `populate_bounds`. 

Now the entire API and structure has changed and I feel like we need to be more cautious about the function use in our codes. I am still working on algorithms for my research right now and cannot spend too much time for these optimizations. But @amberchen122 will help us on the refractory as well. That's why I asked her to look into it